core-api/src/main/java/com/sequenceiq/cloudbreak/validation/ImageCatalogValidator.java

```diff
@@ -0,0 +1,46 @@
+package com.sequenceiq.cloudbreak.validation;
+
+import javax.annotation.PostConstruct;
+import javax.inject.Inject;
+import javax.validation.ConstraintValidator;
+import javax.validation.ConstraintValidatorContext;
+
+import org.springframework.stereotype.Component;
+
+import com.sequenceiq.cloudbreak.validation.bean.ImageCatalogProvider;
+
+import edu.umd.cs.findbugs.annotations.SuppressFBWarnings;
+
+@Component
+public class ImageCatalogValidator implements ConstraintValidator<ValidImageCatalog, String> {
+
+    private static ImageCatalogProvider imageCatalogProvider;
+
+    @Inject
+    private ImageCatalogProvider imageCatalogProviderComponent;
+
+    @PostConstruct
+    @SuppressFBWarnings("ST_WRITE_TO_STATIC_FROM_INSTANCE_METHOD")
+    public void init() {
+        imageCatalogProvider = imageCatalogProviderComponent;
+    }
+
+    @Override
+    public void initialize(ValidImageCatalog constraintAnnotation) {
+    }
+
+    @Override
+    public boolean isValid(String value, ConstraintValidatorContext context) {
+        if (value == null || !value.startsWith("http")) {
+            return false;
+        }
+        try {
+            imageCatalogProvider.getImageCatalogV2(value);
```

Comment:
Isn't this call is too expensive?
Maybe we should use a [head()](https://github.com/hortonworks/cloudbreak/blob/ad6d22a39717b8b199aaf2fd5ca0c85640b76018/cloud-common/src/main/java/com/sequenceiq/cloudbreak/common/service/url/UrlAccessValidationService.java#L37-L36) on the file to validate it's availability.

Reply:
We don't want to validate the availability, we want to check is the content a valid image catalog. We want to be sure to avoid Server Side Request Forgery attacks. In theory it could't be happen because image catalog service also checks it, but it happens lately, when the record is already in the database.

Reply:
In general, Tomi is right, might be better to do a HEAD first, check the size to make sure that it is not extremely large otherwise we might run into an OOM.